### PR TITLE
Add bootstrap to template

### PR DESCRIPTION
**Pull request by dalepotter**

### CODE REVIEW

**Comment by andylolz** [comment]
Cool! [Having done that here](http://please.treadsoft.ly/query/), I keep meaning to send a PR! Glad you have sorted it.

File: templates/index.html [hunk 1]
```diff
@@ -3,191 +3,183 @@
...
+      <div class="header row">
+        <div class="col">
+          <h1 dir="ltr">IATI Data Store CSV Query Builder (Alpha Version)</h1>
+          <p>Please read the <a href="http://datastore.iatistandard.org/docs/user-guide/">user guide</a></p>
+        </div>
+      </div><!-- end class="header" -->
+
+      <div class="query-builder">
+        <div class="row"><!-- result -->
+          {% if api_link %}
```
Comment (hayfield) [inline templates/index.html]: Should there be a matching `endif`?

**Comment by dalepotter** [comment]
Thanks @andylolz - great to look at the link - like the select buttons. I think this PR sees us mostly done with moving over to bootstrap now, but we'll be taking a look at the form inputs and we have ideas for other improvements in coming days too.

File: templates/index.html (hunk 1 above)
Comment (dalepotter) [reply templates/index.html]: Just checked this and the `endif` for this statement is on line 37.  Have fixed the indentation for a nested `if` statement in 0219736 (which previously made it harder to see the `if` / `elseif` / `endif` control flow).

File: custom.css [hunk 2]
```diff
@@ -0,0 +1,10 @@
+.form-control { margin: 10px 0px 0px 0px; }
```
Comment (hayfield) [inline custom.css]: May be worth re-ordering this file to make it a tad easier to skim past logical groupings.

(ofc the file is tiny, so it doesn't really matter if it's unlikely to grow much)

**Comment by andylolz** [comment]
> great to look at the link - like the select buttons.

I probably should have pointed out that it lives here:
https://github.com/andylolz/IATI-Query-Builder/tree/js-query-builder

It’s a shame that I did a bunch of different things in a very small number of commits.

I just took a look – very nice! One thing to note… I think you’re using`class="form-control"` for the control wrapper, rather than `class="form-group"`. `class="form-control"` goes on the control itself. There’s also loads of legacy classes in there that would be worth stripping out.

I know my code has a bunch of unrelated stuff in there, but the HTML could still prove useful.

File: custom.css (hunk 2 above)
Comment (dalepotter) [reply custom.css]: Improved in 2a50356.

**Comment by dalepotter** [comment]
Thanks for the comments @andylolz - missed that `js-query-builder` branch earlier and by the time that I'd taken a look we were mostly there, your work set a good direction for implementation though, so hope you don't feel it was a completely wasted effort...

Thanks for  the tip on `form-control` vs `form-group` - will take a look at that tomorrow.

**Comment by andylolz** [comment]
When you come to fix the `fieldset`, it’s worth bearing in mind #39. That’s the reason I didn’t bother with the “EITHER a country OR a region” thing – it looks bad / is a bad UI, but it’s also inconsistent anyway, because (in order to be consistent) there should be a similar one for reporting org OR reporting org type.

**Comment by dalepotter** [comment]
> When you come to fix the fieldset, it’s worth bearing in mind #39. That’s the reason I didn’t bother with the “EITHER a country OR a region” thing – it looks bad / is a bad UI, but it’s also inconsistent anyway, because (in order to be consistent) there should be a similar one for reporting org OR reporting org type.

Definitely - was doing some wireframe sketching yesterday afternoon and came up with the same idea!

img_20170920_100413533

**Comment by andylolz** [comment]
> Definitely - was doing some wireframe sketching yesterday afternoon and came up with the same idea!

Ace! That’s certainly much nicer.

If you don’t mind a bit of javascript, you could have a radio / select to choose from two options:
 - [ ] country
 - [ ] region

…and then this triggers which chosen select is displayed. That way it’s impossible for the user to do weird, which is better UX (in the “don’t make me think” sort of sense.)

More generally, though: These are all just filters, and most times I guess the user doesn’t care about the majority of these filters. So perhaps they should pick the ones they care about, and the rest should be hidden until needed.

USAID explorer does exactly this really nicely:
https://explorer.usaid.gov/query

**Comment by dalepotter** [comment]
> …and then this triggers which chosen select is displayed. That way it’s impossible for the user to do weird, which is better UX (in the “don’t make me think” sort of sense.)

Nice idea too, I was also thinking about some pre-select 'country' vs. 'region' buttons, although was thinking these could be styled with a graphically too. But radio buttons is definitely more MVP!

I'm going to merge current changes in this PR now, but I've set up a separate issue https://github.com/IATI/IATI-Query-Builder/issues/54 for UI improvements and will copy these comments over.

**Pull request merged**